cpu/fe310/thread_arch.c

```diff
@@ -178,13 +178,23 @@ void cpu_switch_context_exit(void)
     UNREACHABLE();
 }
 
-void thread_yield_higher(void)
+static inline void _ecall_dispatch(uint32_t num, void *ctx)
 {
-    /* Use SW intr to schedule context switch */
-    CLINT_REG(CLINT_MSIP) = 1;
+    /* function arguments are in a0 and a1 as per ABI */
+    __asm__ volatile (
+    "mv a0, %[num] \n"
+    "mv a1, %[ctx] \n"
+    "ECALL\n"
+    "ret"
```

Comment:
This `ret` shouldn't be here as that would cause issues when this function is inlined.

Reply:
And removed